.github/workflows/package-command.yaml

```diff
@@ -2,136 +2,102 @@ name: Package Command
-          {
-            $customProperties = @{ 'RunId'="$runId"; 'PullRequestNumber'="$pullRequestNumber"; 'EventName'="Create Package"; 'IsNewSolution'='$isNewSolution';'SolutionOfferId'='$solutionOfferId';'SolutionSupportedBy'='$solutionSupportedBy';'SolutionName'='$solutionName'; }
-            Send-AppInsightsEventTelemetry -InstrumentationKey $instrumentationKey -EventName "Create Package" -CustomProperties $customProperties
-
-            if ($solutionName -eq '' -or $null -eq $solutionName)
-            {
-              Write-Output "::warning:: Package-Creation : Skipping as Solution name is blank."
-              Send-AppInsightsTraceTelemetry -InstrumentationKey $instrumentationKey -Message "Execution for create-package started and received blank solutionName for Job Run Id : $runId" -Severity Information -CustomProperties $customProperties
-            }
-            else
-            {
-              Send-AppInsightsTraceTelemetry -InstrumentationKey $instrumentationKey -Message "Execution for create-package started for Solution Name : $solutionName, Job Run Id : $runId" -Severity Information -CustomProperties $customProperties
+          token: ${{ env.GeneratedToken}}
 
-              $baseFolderPath = "/home/runner/work/Azure-Sentinel/Azure-Sentinel/"
-              Set-PSRepository PSGallery -InstallationPolicy Trusted
-              Install-Module powershell-yaml
-              ./.script/package-automation/package-generator.ps1 $solutionName $pullRequestNumber $runId $instrumentationKey $defaultPackageVersion $solutionOfferId $baseFolderPath $isNewSolution
-            }
-          }
-          catch
+      - id: checkAutomatedPR
+        name: check-Automated-PR
+        env:
+          BranchName: ${{ github.event.pull_request.head.ref || github.event.client_payload.pull_request.head.ref || github.event.client_payload.pullRequestBranchName }}
+        shell: pwsh
+        run: |
+          $pullRequestBranchName = "$env:BranchName"
+          Write-Host "Pull Request Branch Name is $pullRequestBranchName"
+          $isAutomatedPR = $false
+          if ($pullRequestBranchName -like '*automated-pr')
           {
-            Send-AppInsightsExceptionTelemetry -InstrumentationKey $instrumentationKey -Exception $_.Exception -CustomProperties @{ 'RunId'="$runId"; 'SolutionName'="$solutionName"; 'PullRequestNumber'="$pullRequestNumber"; 'ErrorDetails'="CreatePackage : Error occured in catch block: $_"; 'EventName'="Create Package"; 'SolutionOfferId'="$solutionOfferId"; }
-            exit 1
+            Write-Host "Skipping packaging as it is an automated pr!"
+            $isAutomatedPR = $true
           }
+          Write-Output "isAutomatedPR=$isAutomatedPR" >> $env:GITHUB_OUTPUT
+          Write-Host "Is this Pull Request autogenerated $isAutomatedPR"
+
+      - id: validateAndCreatePackage
+        name: validate-create-package
+        if: ${{ success() && steps.checkAutomatedPR.outputs.isAutomatedPR == 'False' }}
```

Comment:
check the false conditioning, is it case sensitive in above step?

Reply:
verified and its giving the correct branch name

Reply:
![image](https://github.com/Azure/Azure-Sentinel/assets/107389644/80320207-72f8-4227-a922-2106ce34349e)